client/client_test.go

```diff
@@ -4860,6 +4866,184 @@ func testRelativeMountpoint(t *testing.T, sb integration.Sandbox) {
 	require.Equal(t, dt, []byte(id))
 }
 
+// moby/buildkit#2476
+func testFrontendAttrs(t *testing.T, sb integration.Sandbox) {
```

Comment:
Isn't this the same test as `testClientGatewayFrontendAttrs`?

I'm not sure if this is actually a valid case. `Definition` and `FrontendOpt` should be exclusive. Or do you have some use-case? The important case is the frontend build with `gateway.Client.Solve()` . I think that is covered by Dockerfile tests.

Reply:
> Isn't this the same test as `testClientGatewayFrontendAttrs`?

That's the same my bad.

> I'm not sure if this is actually a valid case. `Definition` and `FrontendOpt` should be exclusive. Or do you have some use-case? The important case is the frontend build with `gateway.Client.Solve()` . I think that is covered by Dockerfile tests.

Yes in some extent it could be used programmatically to check frontend metadata, otherwise I don't see any practical use case.